src/code/main.c

```diff
@@ -3,13 +3,13 @@
 
 s32 gScreenWidth = SCREEN_WIDTH;
 s32 gScreenHeight = SCREEN_HEIGHT;
-u32 gSystemHeapSize = 0;
+size_t gSystemHeapSize = 0;
 
 PreNmiBuff* gAppNmiBufferPtr;
 SchedContext gSchedContext;
 PadMgr gPadMgr;
 IrqMgr gIrqMgr;
-u32 gSegments[NUM_SEGMENTS];
+uintptr_t gSegments[NUM_SEGMENTS];
```

Comment:
there are many things to change associated with `gSegments`, like the `SEGMENTED_TO_VIRTUAL` and co macros, and various variables.
idk if it should be done in this PR or in another one

Reply:
I was going to do that in a different PR but I can do it here.